api/src/controllers/websocket_rpc.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+module.exports = WebsocketRpcControllerFactory
+
+const ws = require('ws')
+const debug = require('debug')('ilp-kit:websocket-rpc')
+const Connector = require('../lib/connector')
+
+function WebsocketRpcControllerFactory (deps) {
+  const connector = deps(Connector)
+
+  return class WebsocketRpcController {
+    static init (server) {
+     const socketServer = new ws.Server({
+        perMessageDeflate: false,
+        server
+      })
+
+      debug('listening for websocket connections')
+      socketServer.on('connection', socket => {
+        const url = socket.upgradeReq.url
```

Comment:
you can also use `socketServer.on('connection', (socket, upgradeReq) => {`, see https://github.com/websockets/ws/blob/master/doc/ws.md#event-connection